### PR TITLE
New listener onboarding 

### PR DESCRIPTION
**Pull request by mhpjay422**
Why this PR?
- Updating onboarding flow to new API

### CODE REVIEW

File: lib/flatfile.ts [hunk 1]
```diff
@@ -409,7 +373,7 @@ export const addDocumentToSpace = async (
   return addDocumentResult.data;
 };
 
-export interface BlueprintWithId extends Blueprint {
+export interface BlueprintWithId {
   id: string;
 }
```
Comment (johnmosesman) [inline lib/flatfile.ts:378]: If we're taking away blueprint, this is just `{ id: string}` and doesn't need a custom type

File: lib/job.ts [hunk 2]
```diff
@@ -47,7 +48,7 @@ export const upsertJob = async ({
   return job;
 };
 
-export const validJobRecords = async (records: Record[]) => {
```
Comment (johnmosesman) [inline lib/job.ts:50]: It's time to cut this function, the records we get back should be valid

File: lib/employee-benefit-plan.ts [hunk 3]
```diff
@@ -3,8 +3,11 @@ import { Record } from "./flatfile";
 import { prismaClient } from "./prisma-client";
 import { DateTime } from "luxon";
 import { upsertBenefitPlan } from "./benefit-plan";
+import { RecordsWithLinks } from "@flatfile/api/api";
 
-export const validEmployeeBenefitPlanRecords = async (records: Record[]) => {
```
Comment (johnmosesman) [inline lib/employee-benefit-plan.ts:7]: It's time to cut this function, the records we get back should be valid

File: lib/new-flatfile.ts [hunk 4]
```diff
@@ -72,3 +73,35 @@ export const addGuestToSpace = async ({
     return null;
   }
 };
+
+export const listWorkbooks = async ({ spaceId }: { spaceId: string }) => {
+  try {
+    const flatfile = flatfileClient();
+
+    const result = await flatfile.workbooks.list(
+      spaceId as ListWorkbooksRequest
```
Comment (johnmosesman) [inline lib/new-flatfile.ts:82]: This `as` is odd, what is the actual type expected here?

File: lib/new-flatfile.ts [hunk 5]
```diff
@@ -72,3 +73,35 @@ export const addGuestToSpace = async ({
...
+
+export const getRecords = async ({ sheetId }: { sheetId: string }) => {
+  try {
+    const flatfile = flatfileClient();
+
+    const result = await flatfile.records.get(sheetId);
+
+    console.log("getRecords result", result);
+
+    return result;
```
Comment (johnmosesman) [inline lib/new-flatfile.ts:102]: result or result.data here? What are we actually returning?

File: lib/flatfile.ts (hunk 1 above)
Comment (mhpjay422) [reply lib/flatfile.ts:378]: removed

File: lib/sync-records.ts [hunk 6]
```diff
@@ -25,24 +27,20 @@ export const syncWorkbookRecords = async ({
...
   const jobRecords = await getRecordsByName({
     userId,
-    accessToken,
     workbookName: process.env.ONBOARDING_WORKBOOK_NAME as string,
     sheetName: "Jobs",
     spaceType,
   });
 
-  const totalRecords = employeeRecords.length + jobRecords.length;
+  const totalRecords = employeeRecords!.length + jobRecords!.length;
```
Comment (johnmosesman) [inline lib/sync-records.ts:43]: Are we sure we have records here, or should we be handling this case specifically?

File: lib/sync-records.ts [hunk 7]
```diff
@@ -60,7 +58,9 @@ export const syncWorkbookRecords = async ({
     organizationId,
   });
 
-  const validEmployees = await validEmployeeRecords(employeeRecords);
+  const validEmployees = await validEmployeeRecords(
+    employeeRecords as RecordsWithLinks
+  );
```
Comment (johnmosesman) [inline lib/sync-records.ts:63]: cut these two ^ based on previous comment

File: lib/sync-records.ts [hunk 8]
```diff
@@ -223,32 +220,36 @@ export const syncBenefitPlanRecords = async ({
 
     return;
   }
+  if (employeeBenefitRecords) {
+    const validPlans = await validEmployeeBenefitPlanRecords(
+      employeeBenefitRecords
+    );
```
Comment (johnmosesman) [inline lib/sync-records.ts:226]: cut also

File: lib/job.ts (hunk 2 above)
Comment (mhpjay422) [reply lib/job.ts:50]: removed

File: lib/employee-benefit-plan.ts (hunk 3 above)
Comment (mhpjay422) [reply lib/employee-benefit-plan.ts:7]: removed

File: lib/sync-records.ts (hunk 6 above)
Comment (mhpjay422) [reply lib/sync-records.ts:43]: made code more explicit

File: lib/sync-records.ts (hunk 7 above)
Comment (mhpjay422) [reply lib/sync-records.ts:63]: removed

File: lib/sync-records.ts (hunk 8 above)
Comment (mhpjay422) [reply lib/sync-records.ts:226]: removed

File: lib/new-flatfile.ts (hunk 4 above)
Comment (mhpjay422) [reply lib/new-flatfile.ts:82]: I just needed to wrap in object

File: lib/new-flatfile.ts (hunk 5 above)
Comment (mhpjay422) [reply lib/new-flatfile.ts:102]: The point of the req is to get records. I should dig into the response and actually return records.

File: lib/employee.ts [hunk 9]
```diff
@@ -59,7 +60,7 @@ export const upsertEmployee = async ({
   return employee;
 };
 
-export const validEmployeeRecords = async (records: Record[]) => {
```
Comment (johnmosesman) [inline lib/employee.ts:62]: can cut this

Comment (mhpjay422) [reply lib/employee.ts:62]: removed

**Pull request merged**